### PR TITLE
(feat): add autoupdate functionality

### LINKED ISSUE #5: Feature Request: (Auto) Update Functionality
Hi there.  Will the container be updated for the 8.0.1.577 version update or is there a process to update the client?

Thanks for making this!!

### LINKED ISSUE #31: Upgrade to 8.5?
I was wondering if anyone has tried upgrading to 8.5 yet?

https://www.backblaze.com/blog/announcing-backblaze-computer-backup-v8-5/

### LINKED ISSUE #46: Is backblaze version 8.5.0.659 broken in this wine container?
I first attempted to upgrade my existing container as follows.
first I start an interactive bash shell inside my docker container named "backblaze3"
`docker exec -it --user app backblaze3 /bin/bash`
Then I run the following commands to upgrade
`cd /config/wine/dosdevices/c\:/`
`curl -L "https://www.backblaze.com/win32/install_backblaze.exe" --output "install_backblaze.exe"`
`wine64 "install_backblaze.exe"`

The installer comes up and allows you to run it inside the web gui
after it completes, backblaze ui is broken, it looks like this.
image

I also tried building a new docker container from scratch, when it downloads the latest version of backblaze (8.5.0.659), after it installs, backblaze begins an initial backup, then the UI freezes and becomes unresponsive.  Monitoring of log files indicates nothin is happening.  If you restart the container, the UI looks the same as in the screenshot.

Does anyone know valid links to view the backblaze software version history as well as where to download specific versions?  I want to test the next version back, I'm not sure what version that is.

### PR DESCRIPTION
**Pull request by traktuner**
Scope of this PR:

- [x] ability to auto-update the container on-start
- [x] by default, a known-good version will be used
- [x] by adding the environment variable "FORCE_LATEST_UPDATE=true" the latest available backblaze version will be installed
- [x] pin wine version in the Dockerfiles to get more control over stability
- [x] completely disable the update function with an environment variable

Fixes #5
Fixes #46
Fixes #31

### CODE REVIEW

File: PINNED_BZ_VERSION [hunk 1]
```diff
@@ -0,0 +1,2 @@
+9.0.0.749
+https://web.archive.org/web/20231127163613/https://secure.backblaze.com/win32/install_backblaze.exe
```
Comment (JonathanTreffler) [inline PINNED_BZ_VERSION]: We should probably create a new snapshot before we release the first version with the autoupdater, to get the latest available included in the new docker (and so nobobdy, that used the old manual way to update can be on a version greater than the currently determined stable by us, to make the transition as easy ad possible).

**Review by JonathanTreffler — APPROVED** [review]
This looks good to me, great work.
If you are confident you have tested it enough (I changed it to a brand new installer snapshot, so that could have theoretically broken it) you can merge it yourself and I will create a new release.

**Comment by traktuner** [comment]
Thank you!
I will test it with the new installer again. Should be the same version since this is already the latest one 👍

**Comment by JonathanTreffler** [comment]
> I will test it with the new installer again. Should be the same version since this is already the latest one 👍

You are correct, the new snapshot is also 9.0.0.749
```
$ exiftool install_backblaze.exe 
File Name                       : install_backblaze.exe
Directory                       : .
File Size                       : 29 MB
File Type                       : Win32 EXE
File Type Extension             : exe
MIME Type                       : application/octet-stream
Machine Type                    : Intel 386 or later, and compatibles
Time Stamp                      : 2023:11:03 21:22:12+01:00
Image File Characteristics      : No relocs, Executable, 32-bit
PE Type                         : PE32
Linker Version                  : 14.16
Code Size                       : 258560
Initialized Data Size           : 172544
Uninitialized Data Size         : 0
Entry Point                     : 0xcd3b
OS Version                      : 6.0
Image Version                   : 0.0
Subsystem Version               : 6.0
Subsystem                       : Windows GUI
File Version Number             : 9.0.0.749
Product Version Number          : 9.0.0.749
File Flags Mask                 : 0x003f
File Flags                      : (none)
File OS                         : Win32
Object File Type                : Executable application
File Subtype                    : 0
Language Code                   : English (U.S.)
Character Set                   : Unicode
Company Name                    : Backblaze
File Description                : 
File Version                    : 9.0.0.749
Legal Copyright                 : Copyright © 2007-2023. Backblaze
Product Name                    : Backblaze Backup
Product Version                 : 9.0.0.749
```

**Comment by JonathanTreffler** [comment]
I'll merge it myself, because I want to add some other changes before the release, that don't make sense in this branch, but depend on these changes. But I will wait for your OK in terms of testing for the release.

**Comment by traktuner** [comment]
All good with the new archive.org url!
Go for release 👍

**Pull request merged**